gearsyncd/gearboxparser.cpp

```diff
@@ -0,0 +1,285 @@
+/*
+ * Copyright 2019-2020 Broadcom Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "gearboxparser.h"
+#include "phyparser.h"
+#include <vector>
+
+void GearboxParser::notifyGearboxConfigDone(bool success)
+{
+    swss::ProducerStateTable *p = getProducerStateTable().get();
+
+    swss::FieldValueTuple finish_notice("success", std::to_string(success));
+    std::vector<swss::FieldValueTuple> attrs = { finish_notice };
+
+    p->set("GearboxConfigDone", attrs);
+}
+
+bool GearboxParser::parse()
+{
+    json root;
+
+    try 
+    {
+        root = getJSONRoot();
+    } 
+    catch (const std::exception& e) 
+    {
+        SWSS_LOG_ERROR("JSON root not parseable");
+        return false;
+    }
+
+    json phys, phy, interfaces, interface, val, lanes;
+
+    std::vector<swss::FieldValueTuple> attrs;
+
+    try 
+    {
+        phys = root["phys"];
+        if (phys.size() == 0) 
+      {
+        SWSS_LOG_ERROR("zero-sized 'phys' field in gearbox configuration");
+        return false;
+      }
+    } 
+    catch (const std::exception& e) 
+    {
+        SWSS_LOG_ERROR("unable to read gearbox configuration (invalid format)");
+        return false;
+    }
+
+    for (uint32_t iter = 0; iter < phys.size(); iter++) 
+    {
+        phy = phys[iter];
+        try {
```

Comment:
Opening brace should go on the following line by itself to align with the prevailing style. Please adjust all files accordingly.

Reply:
thanks, good catch.,

Reply:
@jleveque issue addressed